### PR TITLE
Support new websocket server message "threadUpdated"

### PR DESCRIPTION
**Pull request by ofoucherot**
### Description

Add support to a new Room server message `threadResolvedUpdated` that is broadcasted to the room's active users when a thread is resolved.

#### Related issue(s)
https://linear.app/liveblocks/issue/LB-290/resolve-as-first-class-citizen

### CODE REVIEW

**Review by nvie — COMMENTED** [review]
It looks good to me, however, I have one question/suggestion before approving.

I'm wondering if it would be wise to make this event not specific around the `resolved` property, but instead maybe we should make this a `THREAD_UPDATED` (more generic) event.

Asking, because maybe in the future we'll have more properties we'll want to update on threads, and I'm wondering what that would look like. We'd have to extend the enum every time, and maybe we can simply make this event have a `patch` field which can just be `{ resolved: true }` for now.

Approaching this generically might be nice to combine multiple field updates in one operation as well. So rather than having to send three server messages:

    THREAD_RESOLVED_UPDATED true
    THREAD_FOO_UPDATED 1
    THREAD_BAR_UPDATED null

We could simply do one:

    THREAD_UPDATED { resolved: true, foo: 1, bar: null }

I'm not bothered by `metadata` never being part of that patch. AFAIAC that can remain separate.

**Comment by ofoucherot** [comment]
> It looks good to me, however, I have one question/suggestion before approving.
> 
> I'm wondering if it would be wise to make this event not specific around the `resolved` property, but instead maybe we should make this a `THREAD_UPDATED` (more generic) event.
> 
> Asking, because maybe in the future we'll have more properties we'll want to update on threads, and I'm wondering what that would look like. We'd have to extend the enum every time, and maybe we can simply make this event have a `patch` field which can just be `{ resolved: true }` for now.
> 
> Approaching this generically might be nice to combine multiple field updates in one operation as well. So rather than having to send three server messages:
> 
> ```
> THREAD_RESOLVED_UPDATED true
> THREAD_FOO_UPDATED 1
> THREAD_BAR_UPDATED null
> ```
> 
> We could simply do one:
> 
> ```
> THREAD_UPDATED { resolved: true, foo: 1, bar: null }
> ```
> 
> I'm not bothered by `metadata` never being part of that patch. AFAIAC that can remain separate.

Ahah yes, I came to the same conclusion this morning when I had to change the webhook events.

File: packages/liveblocks-core/src/protocol/ServerMsg.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ export enum ServerMsgCode {
   COMMENT_DELETED = 404,
   COMMENT_REACTION_ADDED = 405,
   COMMENT_REACTION_REMOVED = 406,
+  THREAD_RESOLVED_UPDATED = 408,
```
[the remark this answers is not in the document]
Comment (ofoucherot) [reply packages/liveblocks-core/src/protocol/ServerMsg.ts]: yes!

**Review by nvie — APPROVED** [review]
LGTM! 💯

**Pull request merged**